src/services/dexes/sovryn/SovrynPriceFetcher.ts

```diff
@@ -0,0 +1,71 @@
+import {inject, injectable} from 'inversify';
+import path from 'path';
+import {BigNumber, Contract} from 'ethers';
+import {readFileSync} from 'fs';
+import {fileURLToPath} from 'url';
+import {BaseProvider} from '@ethersproject/providers';
+
+import {FeedFetcherInterface} from 'src/types/fetchers.js';
+import Settings from 'src/types/Settings';
+import {ProviderRepository} from '../../../repositories/ProviderRepository.js';
+import {ChainsIds} from '../../../types/ChainsIds.js';
+import {bigIntToFloatingPoint} from '../../../utils/math.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+export interface Price {
+  price: BigNumber;
+  success: boolean;
+}
+
+export type PricesResponse = {
+  prices: Price[];
+  timestamp: BigNumber;
+};
+
+export type PairRequest = {
+  base: string;
+  quote: string;
+  amountInDecimals: number;
+};
+
+/*
+For getting the prices of different of a Sovryn pool the `base` (input token)
+and `quote` (output token), and the `amount` of the input token should be provided.
+
+weBTC-rUSDT:
+  discrepancy: 1
+  precision: 2
+  inputs:
+    - fetcher:
+        name: SovrynPriceFetcher
+        params:
+          base: '0x69fe5cec81d5ef92600c1a0db1f11986ab3758ab'
+          quote: '0xcb46c0ddc60d18efeb0e586c17af6ea36452dae0'
+          amountIdDecimals: 18
+*/
+@injectable()
+export class SovrynPriceFetcher implements FeedFetcherInterface {
+  @inject('Settings') private settings!: Settings;
+  @inject(ProviderRepository) private providerRepository!: ProviderRepository;
+
+  public async apply(pair: PairRequest): Promise<number> {
+    const prices = await this.getPrice(pair);
+
+    const bigIntPrice = prices.prices[0].price.toBigInt();
```

Comment:
you need to check if price is successful

Reply:
Done

Reply:
please check integration example I provided how to check if price is successful

Reply:
use `success`